modules/contract/contracts.js

```diff
@@ -75,27 +74,40 @@ class Contracts {
         await this.deleteContract(args[0]);
     }
 
-    async listEntryPoints(contractPath) {
+    async listEntryPoints(contract) {
+        const tezosNode = this.config.provider;  
         const fs = require('fs');
         const conseiljs = require(CONSEIL_JS);
-        let conseilServer = { 'url': CONSEIL_SERVER_URL, 'apiKey': CONSEIL_SERVER_APIKEY, 'network': TESTNET_NAME };
+
+        let carthagenetConseilServer = { 'url': CONSEIL_SERVER.Carthagenet.url, 'apiKey': CONSEIL_SERVER.Carthagenet.apiKey, 'network': 'carthagenet' };
+        let mainnetConseilServer = { 'url': CONSEIL_SERVER.Mainnet.url, 'apiKey': CONSEIL_SERVER.Mainnet.apiKey, 'network': 'mainnet' };
+
         let contractCode, contractAddress;
-        let contractObj = Helper.findKeyObj(config.contracts, contractPath);
+        let contractObj = Helper.findKeyObj(this.config.contracts, contract);
         if (contractObj) {
             contractAddress = contractObj.pkh;
         }
+
+        if(contract.startsWith('KT')) {
+            contractAddress = contract;
+        }
       
-        if (!contractAddress && !fs.existsSync(contractPath)) {
-            Logger.error(`Couldn't find the contract, please make sure contract label or address is correct!`);
+        if (!contractAddress && !fs.existsSync(contract)) {
+            Logger.error(`Couldn't find the contract, please make sure contract-file-path/contract-label/contract-address is correct.`);
             return;
         }
 
         try {
             if(contractAddress) {
-                const account = await conseiljs.TezosConseilClient.getAccount(conseilServer, conseilServer.network, contractAddress);
-                contractCode = account.script;
+                if(Helper.isCarthagenetNode(tezosNode)) {
+                    const account = await conseiljs.TezosConseilClient.getAccount(carthagenetConseilServer, carthagenetConseilServer.network, contractAddress);
+                    contractCode = account.script;                    
+                } else if(Helper.isMainnetNode(tezosNode)) {
+                    const account = await conseiljs.TezosConseilClient.getAccount(mainnetConseilServer, mainnetConseilServer.network, contractAddress);
+                    contractCode = account.script;                   
+                }
             } else {
-                contractCode = fs.readFileSync(contractPath, 'utf8');
+                contractCode = fs.readFileSync(contract, 'utf8');
```

Comment:
readFileSync always takes a path as argument. 
Name your variables properly to remove the confusion.